src/PSR7/ResponseValidator.php

```diff
@@ -5,74 +5,42 @@
 namespace OpenAPIValidation\PSR7;
 
 use cebe\openapi\spec\OpenApi;
-use OpenAPIValidation\PSR7\Exception\Response\MissedResponseHeader;
-use OpenAPIValidation\PSR7\Exception\Response\ResponseBodyMismatch;
-use OpenAPIValidation\PSR7\Exception\Response\ResponseHeadersMismatch;
-use OpenAPIValidation\PSR7\Exception\Response\UnexpectedResponseContentType;
-use OpenAPIValidation\PSR7\Exception\Response\UnexpectedResponseHeader;
-use OpenAPIValidation\PSR7\Validators\Body;
-use OpenAPIValidation\PSR7\Validators\Headers;
+use OpenAPIValidation\PSR7\Exception\ValidationFailed;
+use OpenAPIValidation\PSR7\Validators\BodyValidator;
+use OpenAPIValidation\PSR7\Validators\HeadersValidator;
+use OpenAPIValidation\PSR7\Validators\ValidatorChain;
 use Psr\Http\Message\ResponseInterface;
-use Throwable;
 
 class ResponseValidator implements ReusableSchema
 {
     /** @var OpenApi */
     protected $openApi;
-    /** @var SpecFinder */
-    protected $finder;
+    /** @var MessageValidator */
+    protected $validator;
 
     public function __construct(OpenApi $schema)
     {
-        $this->openApi = $schema;
-        $this->finder  = new SpecFinder($this->openApi);
+        $this->openApi   = $schema;
+        $finder          = new SpecFinder($this->openApi);
+        $this->validator = new ValidatorChain(
```

Comment:
This class is now very tiny, since all validators are moved away. I hope we can create something like `GuidedServerRequestValidator` which does not perform path matching (aka routing) and accepts PathOperation with arguments and it will be the same tiny class as this one